### PR TITLE
Aliyun: Remove spring-boot dependency

### PR DESCRIPTION
**Pull request by jbonofre**
This PR removes the spring-boot dependency from Iceberg and implement Aliyun OSS Mock using the JDK HTTP server.

### CODE REVIEW

File: aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java [hunk 1]
```diff
@@ -121,8 +116,9 @@ void deleteBucket(String bucketName) throws IOException {
 
     File dir = new File(root, bucket.getName());
     if (Files.walk(dir.toPath()).anyMatch(p -> p.toFile().isFile())) {
-      throw new AliyunOSSMockLocalController.OssException(
-          409, OSSErrorCode.BUCKET_NOT_EMPTY, "The bucket you tried to delete is not empty. ");
+      // throw new OSSException(409, OSSErrorCode.BUCKET_NOT_EMPTY, "The bucket you tried to delete
```
Comment (manuzhang) [inline aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java:119]: why do we keep this commented line?

**Review by findepi — APPROVED** [review]
Thanks!

Removing redundant dependencies is always a good idea!

File: aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java (hunk 1 above)
Comment (findepi) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java:119]: @jbonofre ?

Comment (jbonofre) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java:119]: My bad, I forgot to clean. I will fix that. Thanks !

Comment (jbonofre) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalStore.java]: Cleaned it up :)

**Comment by jbonofre** [comment]
I'm checking why one check failed.

**Comment by jbonofre** [comment]
@findepi @manuzhang I fixed the spotless. Sorry for the inconvenience.

File: aliyun/src/test/java/org/apache/iceberg/aliyun/TestUtility.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public static AliyunOSSExtension initialize() {
     } else {
       LOG.info(
           "Initializing AliyunOSSExtension implementation with default AliyunOSSMockExtension");
-      extension = AliyunOSSMockExtension.builder().silent().build();
```
Comment (manuzhang) [inline aliyun/src/test/java/org/apache/iceberg/aliyun/TestUtility.java:68]: what's the difference without `silent()`?

Comment (jbonofre) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/TestUtility.java:68]: `silent()` was specific to spring-boot to not display the spring-boot banner and extra log.
Now that we are not using spring-boot, but directly implementing the mock using JDK HTTP server, `silent()` is not applicable anymore.

**Comment by jbonofre** [comment]
@rdblue @RussellSpitzer  as discussed together, this is the PR to remove the spring-boot dependency and use JDK HTTP server directly for mocking.

**Comment by findepi** [comment]
@jbonofre there is a conflict, you may want to resolve

**Comment by jbonofre** [comment]
@findepi conflict resolved. Thanks.

**Review by findepi — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
@jbonofre This has conflicts, @findepi you have the ability to merge this if you are done reviewing :)

**Comment by Fokko** [comment]
@jbonofre @RussellSpitzer Took the liberty of fixing this through the GitHub UI. Let's get this in 🚀

**Pull request merged**